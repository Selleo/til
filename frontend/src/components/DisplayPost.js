import React, { useState, useEffect } from 'react'
import { fetchSinglePost } from '../utils'
import { useParams } from 'react-router-dom'
import Post from '../components/Post'

const { REACT_APP_API_URL: API_URL } = process.env

const DisplayPost = () => {
  const [post, setPost] = useState(null)
  const { id } = useParams()

  useEffect(() => {
    const fetchPost = async () => {
      const post = await fetchSinglePost(`${API_URL}/api/posts/`, id)
      setPost(post)
    }
    fetchPost()
  }, [id])

  if (!post) {
    return <p>...loading...</p>
  }

  return <Post key={post.id} post={post} />
}

export default DisplayPost
