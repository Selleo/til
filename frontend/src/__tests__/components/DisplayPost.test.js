import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import DisplayPost from "../../components/DisplayPost";

describe("<DisplayPost/>", () => {
  const history = createMemoryHistory();

  const component = (
    <Router history={history}>
      <DisplayPost />
    </Router>
  );

  it("renders correctly", () => {
    const { asFragment } = render(component);

    expect(asFragment()).toMatchSnapshot();
  });

  it("initializes with `loading` state", () => {
    const { getByText } = render(component);
    const loadingText = getByText(/loading/i);

    expect(loadingText).toBeInTheDocument();
  });
});
