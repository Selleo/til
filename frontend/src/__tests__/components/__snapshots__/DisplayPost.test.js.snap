// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayPost/> renders correctly 1`] = `
<DocumentFragment>
  <p>
    ...loading...
  </p>
</DocumentFragment>
`;
